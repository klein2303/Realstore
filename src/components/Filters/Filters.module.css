.filterSection {
    margin-right: 0;
    padding: 50px;
    border-radius: 10px;
    width: 230px;
}

.mobilefilter {
    display: flex;
    flex-direction: row;
    align-items: flex-end;
    gap: 10px;
}

.filterCategory {
    width: 225px;
}

.h2text {
    font-family: "Montserrat", sans-serif;
    font-weight: 300;
    font-size: 32px;
    color: #424242;
}

.h3text {
    font-family: "Montserrat", sans-serif;
    font-weight: 200;
    font-size: 18px;
    color: #424242;
}

.h5text {
    font-family: "Montserrat", sans-serif;
    font-weight: 400;
    font-size: 16px;
    color: #424242;
}

.h5text:hover {
    font-weight: bold;
}

.line {
    font-family: "Montserrat", sans-serif;
    font-weight: 400;
    font-size: 16px;
    color: #424242;
}

.categoryAndArrow {
    display: flex;
    justify-content: space-between;
    align-items: center;
    cursor: pointer;
    margin-bottom: 20px;
    margin-top: 20px;
}

.arrow {
    font-size: 20px;
}

.sizeBox {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 50px;
    height: 35px;
    border-radius: 10px;
    margin-right: 2%;
    border: 2px solid black;
}

.selectionSection {
    display: flex;
    align-items: center;
    flex-direction: row-reverse;
    justify-content: space-between;
    height: 60px;
    width: 225px;
    cursor: pointer;
}

.clickBox {
    height: 25px;
    width: 25px;
    border: solid 2px black;
    border-radius: 10px;
}

.clickedBox {
    height: 25px;
    width: 25px;
    border: solid 2px black;
    border-radius: 10px;
    background-color: black;
}

.filterStyling {
    color: #424242;
}

@media (max-width: 780px) {
    .closed {
        padding: 10px;
        margin-top: 40px;
    }

    .categoryAndArrow {
        padding-right: 0;
    }

    .selectionSection {
        margin-bottom: 10px;
    }
}
