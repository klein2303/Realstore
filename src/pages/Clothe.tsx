import {useParams} from "react-router-dom";
import ActionBox from "../components/Clothe/ActionBox";
import Navbar from "../components/Navbar/Navbar";
import Footer from "../components/Footer/Footer";
import styles from "../styles/Clothe.module.css";

const Clothe = () => {
    const {id} = useParams<{id: string}>();
    const productId = id ? parseInt(id) : 0; // Ensure id is parsed correctly

    return (
        <>
            <Navbar />
            <main className={styles.page}>
                <ActionBox productId={productId} />
            </main>
            <Footer />
        </>
    );
};

export default Clothe;
