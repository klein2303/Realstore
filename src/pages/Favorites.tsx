import Navbar from "../components/Navbar/Navbar";
import Footer from "../components/Footer/Footer";
import styles from "../styles/Favorites.module.css";
import {Link} from "react-router-dom";
import ClothingsCards from "../components/Scrolling/ClothingsCards/ClothingsCards";
import {Product} from "../models/Product";
import {useProducts} from "../hooks/useProducts";
import {getFavorites} from "../utils/localStorage";
import {useEffect, useState} from "react";

const Favorites = () => {
    const [favorites, setFavorites] = useState<number[]>([]);
    const [favoriteProducts, setFavoriteProducts] = useState<Product[]>([]);
    const {data: products, isLoading, isError} = useProducts();

    useEffect(() => {
        if (!products) {
            return;
        }
        const favorites = getFavorites();
        setFavorites(favorites);
        const currentFavoriteProducts = products.filter((product: Product) => favorites.includes(product.id));
        setFavoriteProducts(currentFavoriteProducts);
    }, [products]);

    if (isLoading) {
        return (
            <section role="status" aria-live="polite">
                Loading...
            </section>
        );
    }
    if (isError) {
        return (
            <section role="alert" aria-live="assertive">
                Error fetching products.
            </section>
        );
    }

    return (
        <>
            <Navbar />
            <main className={styles.favoritepage} role="region" aria-label="Favorite Products">
                <div className={styles.favoritecontainer}>
                    <nav className={styles.path} aria-label="Navigation">
                        <Link className={styles.pathlink} to={"/"}>
                            Home
                        </Link>{" "}
                        / Favorites
                    </nav>
                    <p className={styles.title}>FAVORITES</p>
                    <section className={styles.favoriteitems} role="list" aria-description="List of favorite products">
                        {favorites.length === 0 ? (
                            <section className={styles.message} role="status" aria-live="polite">
                                No products available.
                            </section>
                        ) : (
                            favoriteProducts.map((product: Product) => (
                                <ClothingsCards
                                    key={product.id}
                                    id={product.id}
                                    title={product.title.substring(0, 20) + "..."}
                                    price={product.price}
                                    category={product.category.charAt(0).toUpperCase() + product.category.substring(1)}
                                    image={product.image}
                                    cart={true}
                                    favorite={true}
                                />
                            ))
                        )}
                    </section>
                </div>
            </main>
            <Footer />
        </>
    );
};

export default Favorites;
