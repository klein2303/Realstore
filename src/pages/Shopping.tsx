import Navbar from "../components/Navbar/Navbar";
import Footer from "../components/Footer/Footer";
import styling from "../styles/Shopping.module.css";
import Scrolling from "../components/Scrolling/Scrolling";
import Filters from "../components/Filters/Filters";
import {useEffect, useState} from "react";
import {getFilter} from "../utils/sessionStorage";

export type Filter = {
    category: string;
    values: {
        min?: number;
        max?: number;
    };
};

const Shopping = () => {
    const [selectedFilters, setSelectedFilters] = useState<Filter>({
        category: "",
        values: {min: undefined, max: undefined},
    });
    useEffect(() => {
        if (getFilter().length > 0) {
            setSelectedFilters(getFilter()[0]);
        }
    }, []);
    return (
        <>
            <Navbar />
            <main className={styling.shoppingpage}>
                <Filters selectedFilter={selectedFilters} setSelectedFilters={setSelectedFilters} />
                <Scrolling favorite={true} cart={true} selectedFilter={selectedFilters} />
            </main>
            <Footer />
        </>
    );
};

export default Shopping;
