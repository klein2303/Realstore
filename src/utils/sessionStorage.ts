import {Filter} from "../pages/Shopping";

export type CartItem = {productId: number; quantity: number};

// Helper function to store items in session storage
const setSessionStorageItems = <T>(key: string, items: T[]): void => {
    sessionStorage.setItem(key, JSON.stringify(items));
};

// Helper function to get items from session storage
const getSessionStorageItems = <T>(key: string): T[] => {
    const storedItems = sessionStorage.getItem(key);
    return storedItems ? JSON.parse(storedItems) : [];
};

// Cart-specific functions
export const getCart = (): CartItem[] => {
    return getSessionStorageItems<CartItem>("cart");
};

export const addToCart = (productId: number, quantity: number = 1): void => {
    const cart = getCart();
    const existingItem = cart.find((item) => item.productId === productId);

    if (existingItem) {
        existingItem.quantity += quantity; // Increment quantity
    } else {
        cart.push({productId, quantity});
    }

    setSessionStorageItems("cart", cart);
};

export const clearCart = (): void => {
    setSessionStorageItems("cart", []);
};

export const updateCartQuantity = (productId: number, quantity: number): void => {
    const cart = getCart();
    const index = cart.findIndex((item) => item.productId === productId);

    if (index !== -1) {
        // Product exists in the cart
        if (quantity > 0) {
            cart[index].quantity = quantity; // Update quantity
        } else {
            cart.splice(index, 1); // Remove if quantity is 0
        }
    } else if (quantity > 0) {
        // Product doesn't exist in the cart, add it
        cart.push({productId, quantity});
    }

    setSessionStorageItems("cart", cart); // Save updated cart to sessionStorage
};

// Filtered items-specific functions
export const getFilter = (): Filter[] => {
    return getSessionStorageItems<Filter>("filter");
};

export const setFilter = (filter: Filter[]): void => {
    setSessionStorageItems("filter", filter);
};

export const clearFilter = (): void => {
    setSessionStorageItems("filter", []);
};
