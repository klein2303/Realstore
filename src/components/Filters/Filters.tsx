import {useState, useEffect} from "react";
import styles from "./Filters.module.css";
import {IoIosArrowForward, IoIosArrowDown} from "react-icons/io";
import {Filter} from "../../pages/Shopping.tsx";
import {setFilter} from "../../utils/sessionStorage.ts";

type filterTabs = {
    filter: boolean;
    category: boolean;
    priceRange: boolean;
};

interface FiltersProps {
    selectedFilter: Filter;
    setSelectedFilters: (filter: Filter) => void;
}
const Filters = ({selectedFilter, setSelectedFilters}: FiltersProps) => {
    const [filterTabs, setFilterTabs] = useState<filterTabs>({filter: false, category: false, priceRange: false});
    const [isMobile, setIsMobile] = useState(window.innerWidth <= 780);
    const categories = ["men's clothing", "women's clothing", "jewelery", "electronics"];
    const values = ["0-10", "10-50", "50-100", "100-500", "500-1000"];

    const toggleFiltersSection = () =>
        setFilterTabs({filter: !filterTabs.filter, category: filterTabs.category, priceRange: filterTabs.priceRange});

    const toggleCategoriesSection = () => {
        setFilterTabs({filter: filterTabs.filter, category: !filterTabs.category, priceRange: filterTabs.priceRange});
    };

    const togglePriceRangeSection = () => {
        setFilterTabs({filter: filterTabs.filter, category: filterTabs.category, priceRange: !filterTabs.priceRange});
    };

    const handleFilterChange = (category: string, min?: number, max?: number) => {
        const filter: Filter = {
            category: category,
            values: {
                min: min,
                max: max,
            },
        };
        setFilter([filter]); // Session storage
        setSelectedFilters(filter);
    };

    const handleResize = () => {
        setIsMobile(window.innerWidth <= 780);
    };

    const capitalize = (text: string) => {
        return text.charAt(0).toUpperCase() + text.substring(1);
    };

    useEffect(() => {
        window.addEventListener("resize", handleResize);
        return () => window.removeEventListener("resize", handleResize);
    }, []);

    return (
        <section className={filterTabs.filter ? styles.filterSection : styles.closed} aria-label="Filter Options">
            <header className={styles.categoryAndArrow}>
                {isMobile && (
                    <div
                        className={styles.mobilefilter}
                        onClick={toggleFiltersSection}
                        role="button"
                        aria-expanded={filterTabs.filter}
                        tabIndex={0}
                        aria-controls="filter-options"
                        onKeyDown={(e) => e.key === "Enter" && toggleFiltersSection()}
                        aria-label="Toggle Filters Section">
                        <h2 className={styles.h2text}>Filters</h2>
                        <span className={styles.arrow}>
                            {filterTabs.filter ? <IoIosArrowDown /> : <IoIosArrowForward />}
                        </span>
                    </div>
                )}
            </header>
            {!isMobile && <h2 className={styles.h2text}>Filters</h2>}
            {(!isMobile || filterTabs.filter) && (
                <>
                    <section className={styles.filterCategory}>
                        <header
                            className={styles.categoryAndArrow}
                            onClick={toggleCategoriesSection}
                            role="button"
                            aria-expanded={filterTabs.category}
                            aria-controls="category-options"
                            tabIndex={0}
                            onKeyDown={(e) => e.key === "Enter" && toggleCategoriesSection()}
                            aria-label="Toggle Categories Section">
                            <h3 className={styles.h3text}>Categories</h3>
                            <span className={styles.arrow}>
                                {filterTabs.category ? <IoIosArrowDown /> : <IoIosArrowForward />}
                            </span>
                        </header>
                        {filterTabs.category && (
                            <ul id="category-options" aria-label="Filter by Category" className={styles.filterStyling}>
                                {categories.map((title) => {
                                    let value = "";
                                    if (selectedFilter.category !== title) {
                                        value = title;
                                    }
                                    return (
                                        <li
                                            key={title}
                                            className={styles.selectionSection}
                                            role="button"
                                            tabIndex={0}
                                            onClick={() =>
                                                handleFilterChange(
                                                    value,
                                                    selectedFilter.values.min,
                                                    selectedFilter.values.max,
                                                )
                                            }
                                            onKeyDown={(e) =>
                                                e.key === "Enter" &&
                                                handleFilterChange(
                                                    value,
                                                    selectedFilter.values.min,
                                                    selectedFilter.values.max,
                                                )
                                            }
                                            aria-label={`Filter by ${capitalize(title)}`}>
                                            <button
                                                className={
                                                    selectedFilter.category == title
                                                        ? styles.clickedBox
                                                        : styles.clickBox
                                                }
                                                aria-pressed={selectedFilter.category === title}
                                                aria-label={
                                                    selectedFilter.category === title
                                                        ? `Selected ${capitalize(title)}`
                                                        : `Select ${capitalize(title)}`
                                                }
                                            />
                                            <h5 className={styles.line}>{capitalize(title)}</h5>
                                        </li>
                                    );
                                })}
                            </ul>
                        )}
                        <h5 className={styles.h5text}>------------------------------------------</h5>
                    </section>
                    <section className={styles.filterCategory}>
                        <header
                            className={styles.categoryAndArrow}
                            onClick={togglePriceRangeSection}
                            role="button"
                            aria-expanded={filterTabs.priceRange}
                            aria-controls="price-range-options"
                            tabIndex={0}
                            onKeyDown={(e) => e.key === "Enter" && togglePriceRangeSection()}
                            aria-label="Toggle Price Range Section">
                            <h3 className={styles.h3text}>Price Range</h3>
                            <span className={styles.arrow}>
                                {filterTabs.priceRange ? <IoIosArrowDown /> : <IoIosArrowForward />}
                            </span>
                        </header>
                        {filterTabs.priceRange && (
                            <ul
                                id="price-range-options"
                                aria-label="Filter by Price Range"
                                className={styles.filterStyling}>
                                {values.map((value: string) => {
                                    const values = value.split("-");
                                    let min: number | undefined = parseInt(values[0]);
                                    let max: number | undefined = parseInt(values[1]);
                                    const selectedFilterValue = `${selectedFilter.values.min}-${selectedFilter.values.max}`;
                                    if (selectedFilterValue == value) {
                                        min = undefined;
                                        max = undefined;
                                    }
                                    return (
                                        <li
                                            key={value}
                                            className={styles.selectionSection}
                                            role="button"
                                            tabIndex={0}
                                            onClick={() => handleFilterChange(selectedFilter.category, min, max)}
                                            onKeyDown={(e) =>
                                                e.key === "Enter" &&
                                                handleFilterChange(selectedFilter.category, min, max)
                                            }
                                            aria-label={`Filter by price range ${value} dollars`}>
                                            <button
                                                className={
                                                    selectedFilterValue == value ? styles.clickedBox : styles.clickBox
                                                }
                                                aria-pressed={selectedFilterValue === value}
                                                aria-label={
                                                    selectedFilterValue === value
                                                        ? `Selected price range ${value}`
                                                        : `Select price range ${value}`
                                                }
                                            />
                                            <h5 className={styles.h5text}>{value} $</h5>
                                        </li>
                                    );
                                })}
                            </ul>
                        )}
                        <h5 className={styles.h5text}>------------------------------------------</h5>
                    </section>
                </>
            )}
        </section>
    );
};

export default Filters;
