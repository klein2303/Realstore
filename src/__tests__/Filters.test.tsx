import {render, screen, fireEvent} from "@testing-library/react";
import {describe, it, expect, beforeEach, vi} from "vitest";
import Filters from "../components/Filters/Filters";
import {setFilter} from "../utils/sessionStorage";
import {Filter} from "../pages/Shopping";

vi.mock("../utils/sessionStorage", () => ({
    setFilter: vi.fn(),
}));

describe("Filters component", () => {
    const mockSetSelectedFilters = vi.fn();

    beforeEach(() => {
        vi.resetAllMocks();
        mockSetSelectedFilters.mockClear();
        Object.defineProperty(window, "innerWidth", {writable: true, configurable: true, value: 1024});
    });

    it("renders correctly in desktop view", () => {
        const initialFilter: Filter = {
            category: "",
            values: {
                min: undefined,
                max: undefined,
            },
        };

        render(<Filters selectedFilter={initialFilter} setSelectedFilters={mockSetSelectedFilters} />);

        expect(screen.getByText(/Filters/i)).toBeInTheDocument();
        expect(screen.getByText(/Categories/i)).toBeInTheDocument();
        expect(screen.getByText(/Price Range/i)).toBeInTheDocument();
    });

    it("toggles filter sections correctly in mobile view", () => {
        Object.defineProperty(window, "innerWidth", {value: 500});
        window.dispatchEvent(new Event("resize"));

        const initialFilter: Filter = {
            category: "",
            values: {
                min: undefined,
                max: undefined,
            },
        };

        render(<Filters selectedFilter={initialFilter} setSelectedFilters={mockSetSelectedFilters} />);

        const filtersHeader = screen.getByRole("button", {name: /toggle filters section/i});
        expect(filtersHeader).toBeInTheDocument();

        fireEvent.click(filtersHeader);
        expect(screen.getByText(/Categories/i)).toBeInTheDocument();
    });

    it("calls setFilter and setSelectedFilters when a category is selected", () => {
        const initialFilter: Filter = {
            category: "",
            values: {
                min: undefined,
                max: undefined,
            },
        };

        render(<Filters selectedFilter={initialFilter} setSelectedFilters={mockSetSelectedFilters} />);

        const categoriesHeader = screen.getByRole("button", {name: /toggle categories section/i});
        fireEvent.click(categoriesHeader);

        const categoryButton = screen.getByRole("button", {name: /select Men's clothing/i});
        fireEvent.click(categoryButton);

        expect(setFilter).toHaveBeenCalledWith([
            {
                category: "men's clothing",
                values: {min: undefined, max: undefined},
            },
        ]);
        expect(mockSetSelectedFilters).toHaveBeenCalledWith({
            category: "men's clothing",
            values: {min: undefined, max: undefined},
        });
    });

    it("calls setFilter and setSelectedFilters when a price range is selected", () => {
        const initialFilter: Filter = {
            category: "",
            values: {
                min: undefined,
                max: undefined,
            },
        };

        render(<Filters selectedFilter={initialFilter} setSelectedFilters={mockSetSelectedFilters} />);

        const priceRangeHeader = screen.getByRole("button", {name: /toggle price range section/i});
        fireEvent.click(priceRangeHeader);

        const priceRangeButton = screen.getByRole("button", {name: /select price range 10-50/i});
        fireEvent.click(priceRangeButton);

        expect(setFilter).toHaveBeenCalledWith([
            {
                category: "",
                values: {min: 10, max: 50},
            },
        ]);
        expect(mockSetSelectedFilters).toHaveBeenCalledWith({
            category: "",
            values: {min: 10, max: 50},
        });
    });

    it("renders correctly on resize", () => {
        const initialFilter: Filter = {
            category: "",
            values: {
                min: undefined,
                max: undefined,
            },
        };

        Object.defineProperty(window, "innerWidth", {writable: true, configurable: true, value: 500});
        window.dispatchEvent(new Event("resize"));

        render(<Filters selectedFilter={initialFilter} setSelectedFilters={mockSetSelectedFilters} />);

        const filtersHeader = screen.getByRole("button", {name: /toggle filters section/i});
        expect(filtersHeader).toBeInTheDocument();

        fireEvent.click(filtersHeader);

        expect(screen.getByText(/Categories/i)).toBeInTheDocument();
    });

    it("matches snapshot for desktop view", () => {
        const initialFilter: Filter = {
            category: "",
            values: {
                min: undefined,
                max: undefined,
            },
        };

        const {container} = render(
            <Filters selectedFilter={initialFilter} setSelectedFilters={mockSetSelectedFilters} />,
        );

        expect(container).toMatchSnapshot();
    });

    it("matches snapshot for mobile view", () => {
        Object.defineProperty(window, "innerWidth", {value: 500});
        window.dispatchEvent(new Event("resize"));

        const initialFilter: Filter = {
            category: "",
            values: {
                min: undefined,
                max: undefined,
            },
        };

        const {container} = render(
            <Filters selectedFilter={initialFilter} setSelectedFilters={mockSetSelectedFilters} />,
        );

        expect(container).toMatchSnapshot();
    });

    it("matches snapshot when categories are expanded in mobile view", () => {
        const initialFilter: Filter = {
            category: "",
            values: {
                min: undefined,
                max: undefined,
            },
        };

        // Set the viewport to mobile dimensions
        Object.defineProperty(window, "innerWidth", {writable: true, configurable: true, value: 500});
        window.dispatchEvent(new Event("resize"));

        // Render the component
        const {container} = render(
            <Filters selectedFilter={initialFilter} setSelectedFilters={mockSetSelectedFilters} />,
        );

        // Find and interact with the "Toggle Filters Section" button
        const filtersHeader = screen.getByRole("button", {name: /toggle filters section/i});
        fireEvent.click(filtersHeader);

        // Expand the categories section
        const categoryHeader = screen.getByRole("button", {name: /toggle categories section/i});
        fireEvent.click(categoryHeader);

        // Match snapshot
        expect(container).toMatchSnapshot();
    });

    it("matches snapshot when price range is expanded", () => {
        const initialFilter: Filter = {
            category: "",
            values: {
                min: undefined,
                max: undefined,
            },
        };

        // Render the component
        const {container} = render(
            <Filters selectedFilter={initialFilter} setSelectedFilters={mockSetSelectedFilters} />,
        );

        // For mobile view, toggle filters section first
        const filtersHeader = screen.queryByRole("button", {name: /toggle filters section/i});
        if (filtersHeader) {
            fireEvent.click(filtersHeader);
        }

        // Expand the price range section
        const priceRangeHeader = screen.getByRole("button", {name: /toggle price range section/i});
        fireEvent.click(priceRangeHeader);

        // Match the snapshot
        expect(container).toMatchSnapshot();
    });
});
