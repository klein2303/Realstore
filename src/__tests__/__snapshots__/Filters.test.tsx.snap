// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Filters component > matches snapshot for desktop view 1`] = `
<div>
  <section
    aria-label="Filter Options"
    class="_closed_950627"
  >
    <header
      class="_categoryAndArrow_950627"
    />
    <h2
      class="_h2text_950627"
    >
      Filters
    </h2>
    <section
      class="_filterCategory_950627"
    >
      <header
        aria-controls="category-options"
        aria-expanded="false"
        aria-label="Toggle Categories Section"
        class="_categoryAndArrow_950627"
        role="button"
        tabindex="0"
      >
        <h3
          class="_h3text_950627"
        >
          Categories
        </h3>
        <span
          class="_arrow_950627"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M294.1 256L167 129c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.3 34 0L345 239c9.1 9.1 9.3 23.7.7 33.1L201.1 417c-4.7 4.7-10.9 7-17 7s-12.3-2.3-17-7c-9.4-9.4-9.4-24.6 0-33.9l127-127.1z"
            />
          </svg>
        </span>
      </header>
      <h5
        class="_h5text_950627"
      >
        ------------------------------------------
      </h5>
    </section>
    <section
      class="_filterCategory_950627"
    >
      <header
        aria-controls="price-range-options"
        aria-expanded="false"
        aria-label="Toggle Price Range Section"
        class="_categoryAndArrow_950627"
        role="button"
        tabindex="0"
      >
        <h3
          class="_h3text_950627"
        >
          Price Range
        </h3>
        <span
          class="_arrow_950627"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M294.1 256L167 129c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.3 34 0L345 239c9.1 9.1 9.3 23.7.7 33.1L201.1 417c-4.7 4.7-10.9 7-17 7s-12.3-2.3-17-7c-9.4-9.4-9.4-24.6 0-33.9l127-127.1z"
            />
          </svg>
        </span>
      </header>
      <h5
        class="_h5text_950627"
      >
        ------------------------------------------
      </h5>
    </section>
  </section>
</div>
`;

exports[`Filters component > matches snapshot for mobile view 1`] = `
<div>
  <section
    aria-label="Filter Options"
    class="_closed_950627"
  >
    <header
      class="_categoryAndArrow_950627"
    >
      <div
        aria-controls="filter-options"
        aria-expanded="false"
        aria-label="Toggle Filters Section"
        class="_mobilefilter_950627"
        role="button"
        tabindex="0"
      >
        <h2
          class="_h2text_950627"
        >
          Filters
        </h2>
        <span
          class="_arrow_950627"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M294.1 256L167 129c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.3 34 0L345 239c9.1 9.1 9.3 23.7.7 33.1L201.1 417c-4.7 4.7-10.9 7-17 7s-12.3-2.3-17-7c-9.4-9.4-9.4-24.6 0-33.9l127-127.1z"
            />
          </svg>
        </span>
      </div>
    </header>
  </section>
</div>
`;

exports[`Filters component > matches snapshot when categories are expanded in mobile view 1`] = `
<div>
  <section
    aria-label="Filter Options"
    class="_filterSection_950627"
  >
    <header
      class="_categoryAndArrow_950627"
    >
      <div
        aria-controls="filter-options"
        aria-expanded="true"
        aria-label="Toggle Filters Section"
        class="_mobilefilter_950627"
        role="button"
        tabindex="0"
      >
        <h2
          class="_h2text_950627"
        >
          Filters
        </h2>
        <span
          class="_arrow_950627"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 294.1L383 167c9.4-9.4 24.6-9.4 33.9 0s9.3 24.6 0 34L273 345c-9.1 9.1-23.7 9.3-33.1.7L95 201.1c-4.7-4.7-7-10.9-7-17s2.3-12.3 7-17c9.4-9.4 24.6-9.4 33.9 0l127.1 127z"
            />
          </svg>
        </span>
      </div>
    </header>
    <section
      class="_filterCategory_950627"
    >
      <header
        aria-controls="category-options"
        aria-expanded="true"
        aria-label="Toggle Categories Section"
        class="_categoryAndArrow_950627"
        role="button"
        tabindex="0"
      >
        <h3
          class="_h3text_950627"
        >
          Categories
        </h3>
        <span
          class="_arrow_950627"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 294.1L383 167c9.4-9.4 24.6-9.4 33.9 0s9.3 24.6 0 34L273 345c-9.1 9.1-23.7 9.3-33.1.7L95 201.1c-4.7-4.7-7-10.9-7-17s2.3-12.3 7-17c9.4-9.4 24.6-9.4 33.9 0l127.1 127z"
            />
          </svg>
        </span>
      </header>
      <ul
        aria-label="Filter by Category"
        class="_filterStyling_950627"
        id="category-options"
      >
        <li
          aria-label="Filter by Men's clothing"
          class="_selectionSection_950627"
          role="button"
          tabindex="0"
        >
          <button
            aria-label="Select Men's clothing"
            aria-pressed="false"
            class="_clickBox_950627"
          />
          <h5
            class="_line_950627"
          >
            Men's clothing
          </h5>
        </li>
        <li
          aria-label="Filter by Women's clothing"
          class="_selectionSection_950627"
          role="button"
          tabindex="0"
        >
          <button
            aria-label="Select Women's clothing"
            aria-pressed="false"
            class="_clickBox_950627"
          />
          <h5
            class="_line_950627"
          >
            Women's clothing
          </h5>
        </li>
        <li
          aria-label="Filter by Jewelery"
          class="_selectionSection_950627"
          role="button"
          tabindex="0"
        >
          <button
            aria-label="Select Jewelery"
            aria-pressed="false"
            class="_clickBox_950627"
          />
          <h5
            class="_line_950627"
          >
            Jewelery
          </h5>
        </li>
        <li
          aria-label="Filter by Electronics"
          class="_selectionSection_950627"
          role="button"
          tabindex="0"
        >
          <button
            aria-label="Select Electronics"
            aria-pressed="false"
            class="_clickBox_950627"
          />
          <h5
            class="_line_950627"
          >
            Electronics
          </h5>
        </li>
      </ul>
      <h5
        class="_h5text_950627"
      >
        ------------------------------------------
      </h5>
    </section>
    <section
      class="_filterCategory_950627"
    >
      <header
        aria-controls="price-range-options"
        aria-expanded="false"
        aria-label="Toggle Price Range Section"
        class="_categoryAndArrow_950627"
        role="button"
        tabindex="0"
      >
        <h3
          class="_h3text_950627"
        >
          Price Range
        </h3>
        <span
          class="_arrow_950627"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M294.1 256L167 129c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.3 34 0L345 239c9.1 9.1 9.3 23.7.7 33.1L201.1 417c-4.7 4.7-10.9 7-17 7s-12.3-2.3-17-7c-9.4-9.4-9.4-24.6 0-33.9l127-127.1z"
            />
          </svg>
        </span>
      </header>
      <h5
        class="_h5text_950627"
      >
        ------------------------------------------
      </h5>
    </section>
  </section>
</div>
`;

exports[`Filters component > matches snapshot when price range is expanded 1`] = `
<div>
  <section
    aria-label="Filter Options"
    class="_closed_950627"
  >
    <header
      class="_categoryAndArrow_950627"
    />
    <h2
      class="_h2text_950627"
    >
      Filters
    </h2>
    <section
      class="_filterCategory_950627"
    >
      <header
        aria-controls="category-options"
        aria-expanded="false"
        aria-label="Toggle Categories Section"
        class="_categoryAndArrow_950627"
        role="button"
        tabindex="0"
      >
        <h3
          class="_h3text_950627"
        >
          Categories
        </h3>
        <span
          class="_arrow_950627"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M294.1 256L167 129c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.3 34 0L345 239c9.1 9.1 9.3 23.7.7 33.1L201.1 417c-4.7 4.7-10.9 7-17 7s-12.3-2.3-17-7c-9.4-9.4-9.4-24.6 0-33.9l127-127.1z"
            />
          </svg>
        </span>
      </header>
      <h5
        class="_h5text_950627"
      >
        ------------------------------------------
      </h5>
    </section>
    <section
      class="_filterCategory_950627"
    >
      <header
        aria-controls="price-range-options"
        aria-expanded="true"
        aria-label="Toggle Price Range Section"
        class="_categoryAndArrow_950627"
        role="button"
        tabindex="0"
      >
        <h3
          class="_h3text_950627"
        >
          Price Range
        </h3>
        <span
          class="_arrow_950627"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 294.1L383 167c9.4-9.4 24.6-9.4 33.9 0s9.3 24.6 0 34L273 345c-9.1 9.1-23.7 9.3-33.1.7L95 201.1c-4.7-4.7-7-10.9-7-17s2.3-12.3 7-17c9.4-9.4 24.6-9.4 33.9 0l127.1 127z"
            />
          </svg>
        </span>
      </header>
      <ul
        aria-label="Filter by Price Range"
        class="_filterStyling_950627"
        id="price-range-options"
      >
        <li
          aria-label="Filter by price range 0-10 dollars"
          class="_selectionSection_950627"
          role="button"
          tabindex="0"
        >
          <button
            aria-label="Select price range 0-10"
            aria-pressed="false"
            class="_clickBox_950627"
          />
          <h5
            class="_h5text_950627"
          >
            0-10
             $
          </h5>
        </li>
        <li
          aria-label="Filter by price range 10-50 dollars"
          class="_selectionSection_950627"
          role="button"
          tabindex="0"
        >
          <button
            aria-label="Select price range 10-50"
            aria-pressed="false"
            class="_clickBox_950627"
          />
          <h5
            class="_h5text_950627"
          >
            10-50
             $
          </h5>
        </li>
        <li
          aria-label="Filter by price range 50-100 dollars"
          class="_selectionSection_950627"
          role="button"
          tabindex="0"
        >
          <button
            aria-label="Select price range 50-100"
            aria-pressed="false"
            class="_clickBox_950627"
          />
          <h5
            class="_h5text_950627"
          >
            50-100
             $
          </h5>
        </li>
        <li
          aria-label="Filter by price range 100-500 dollars"
          class="_selectionSection_950627"
          role="button"
          tabindex="0"
        >
          <button
            aria-label="Select price range 100-500"
            aria-pressed="false"
            class="_clickBox_950627"
          />
          <h5
            class="_h5text_950627"
          >
            100-500
             $
          </h5>
        </li>
        <li
          aria-label="Filter by price range 500-1000 dollars"
          class="_selectionSection_950627"
          role="button"
          tabindex="0"
        >
          <button
            aria-label="Select price range 500-1000"
            aria-pressed="false"
            class="_clickBox_950627"
          />
          <h5
            class="_h5text_950627"
          >
            500-1000
             $
          </h5>
        </li>
      </ul>
      <h5
        class="_h5text_950627"
      >
        ------------------------------------------
      </h5>
    </section>
  </section>
</div>
`;
